### ISSUE TITLE
Wrong serialization result for rgba/hsla spaces

### ISSUE THREAD

**Issue by kajkal** [color-js/color.js#296]
There is this bug:
```js
const c = new Color("rgba(255,0,0,1)");
c.toString({format: "rgba"}); // rgba(100%, 0%, 0%), expected rgba(100%, 0%, 0%, 1)
```
the fix is to check `format.lastAlpha` flag in [the serialize.js function](https://github.com/LeaVerou/color.js/blob/cf124a4931914a2ac3bf33988fdf8e73170bd5bb/src/serialize.js#L80):
```js
let strAlpha = (format.lastAlpha || (alpha < 1 && !format.noAlpha))
    ? `${format.commas? "," : " /"} ${alpha}`
    : "";
```

However, is it necessary to have two flags to determine whether a given format should include alpha?
To my knowledge, there are only 3 cases:
1) alpha is required (`rgba`, `hsla`)
2) alpha is optional (`alpha < 1` `rgb`/`hsl` for CSS Color 4)
3) alpha is ignored (`rgb_number`)

So the final flag should have 3 values, this can be achieved by, for example: 
```ts
interface Format {
    /**
     * true - alpha always present in serialized string
     * undefined - default, alpha present when < 1
     * false - no alpha in serialized string
     */
    withAlpha?: boolean | undefined;
    // or more descriptively by string union:
    withAlpha?: "always" | "auto" | "never";
}
```
```
{ noAlpha: true } => { withAlpha: false }
{ lastAlpha: true } => { withAlpha: true }
```

Is there something I don't know about or can I make PR to merge `noAlpha` and `lastAlpha` flags into one?

**Comment by LeaVerou** [comment]
I like this idea. I'd rename the flag to simply `alpha` and use the enums `"always"`, `"never"`, `"auto"`.

**Comment by svgeesus** [comment]
> alpha is required (`rgba`, `hsla`)

Alpha is [never required](https://drafts.csswg.org/css-color-4/#rgb-functions) even for `rgba()` and `hsla()`, although it used to be (CSS Color 3 and earlier). This is true of both the legacy (comma-separated) and modern syntactic forms.

**Comment by LeaVerou** [comment]
This has now been implemented and will be in the upcoming v0.6.0 release. You can use `alpha: true` to force alpha even when it's 100%, `alpha: false` to prevent alpha from showing up, and `commas: true` to force commas. These together should give you the result you were looking for, even though as @svgeesus said, the current output is perfectly valid.
